File: src/data/F2.java
package data;

import java.util.Arrays;

public class F2 implements Runnable {
    private final int N;
    private final Data.Method method;
    int[][] ML, MH, MG;

    public F2(int n, Data.Method method) {
        this.N = n;
        this.method = method;
    }

    @Override
    public void run() {
        Data d = new Data(N);
        if (N > 1000) {
            switch (method) {
                case File -> {
                    ML = d.getFiledMatrix("asset/ML.txt");
                    MH = d.getFiledMatrix("asset/MH.txt");
                    MG = d.getFiledMatrix("asset/MG.txt");
                }
                case Number -> {
                    ML = d.getNumberedMatrix(1);
                    MH = d.getNumberedMatrix(1);
                    MG = d.getNumberedMatrix(1);
                }
                case Random -> {
                    ML = d.getRandomizedMatrix();
                    MH = d.getRandomizedMatrix();
                    MG = d.getRandomizedMatrix();
                }
                default -> throw new Error();
            }
        } else {
            // keyboard
        }
        int[][] MK = Data.matricesSum(ML, Data.matricesMulti(MH, MG));
        System.out.println("MK = " + Arrays.deepToString(MK));
    }
}

File: python/data/f2.py
from data import Data
import numpy as np


def f2(comm, n, method):
    data = Data(n)
    if n > 1000:
        if method is Data.Method.FILE:
            ml = data.get_file_matrix('asset/ML.txt')
            mh = data.get_file_matrix('asset/MH.txt')
            mg = data.get_file_matrix('asset/MG.txt')
        elif method is Data.Method.NUMBER:
            ml = data.get_number_matrix(2)
            mh = data.get_number_matrix(2)
            mg = data.get_number_matrix(2)
        elif method is Data.Method.RANDOM:
            ml = data.get_random_matrix()
            mh = data.get_random_matrix()
            mg = data.get_random_matrix()
        else:
            raise Exception("Incorrect method.")
    elif n < 5:
        print "f2 reading"
        ml = data.get_keyboard_matrix()
        mh = data.get_keyboard_matrix()
        mg = data.get_keyboard_matrix()
        comm.Barrier()
    else:
        raise Exception("N should be less than 5 or more that 1000.")

    mk = ml + np.dot(mh, mg)
    print 'MK =\n', mk
